akka-docs/src/test/scala/docs/stream/cookbook/RecipeByteStrings.scala

```diff
@@ -23,14 +23,12 @@ class RecipeByteStrings extends RecipeSpec {
         val in = Inlet[ByteString]("Chunker.in")
         val out = Outlet[ByteString]("Chunker.out")
         override val shape = FlowShape.of(in, out)
-
         override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
           private var buffer = ByteString.empty
 
           setHandler(out, new OutHandler {
             override def onPull(): Unit = {
-              if (isClosed(in)) emitChunk()
-              else pull(in)
```

Comment:
Maybe pull if the buffer has less than one chunk in it rather?

Reply:
I had that initially but then if the upstream doesn't then push data as there isn't any then we could have pushed data that now remains in the buffer. Or do you mean push the buffer and do a pull?

Reply:
ah, sorry, I got it wrong and thought it never emits anything but chunk size, while what it does is emit up to chunk size but not larger?

Reply:
That is what it does, i think we'd need a timer otherwise or a smaller chink could remain buffered for every.

In the docs we call this:

"Given a stream of ByteString s we want to produce a stream of ByteString s containing the same bytes in the same sequence, but capping the size of ByteString s. In other words we want to slice up ByteString s into smaller chunks if they exceed a size threshold."

Reply:
👍